### PR TITLE
feat: comprehensive testing framework and enterprise CI/CD pipeline

### LINKED ISSUE #5: Generate Go SDK from proto
## Description
Set up automated generation of Go SDK bindings from protobuf definitions and integrate into CI pipeline.

## Tasks
- [ ] Integrate `buf generate` into CI/CD pipeline
- [ ] Produce Go bindings in sdk/go/ directory
- [ ] Add example usage code in README
- [ ] Set up automated testing of generated code

## Acceptance Criteria
- CI automatically generates Go code on proto changes
- Generated code is placed in sdk/go/ directory
- README includes clear usage examples
- Generated code passes all tests

### PR DESCRIPTION
**Pull request by rshade**
Implement production-ready testing framework, CI/CD pipeline, and cross-vendor examples for PulumiCost specification v0.1.0.

## Summary

- **Complete Testing Framework**: In-memory gRPC testing with multi-level conformance tests
- **Enterprise CI/CD Pipeline**: Automated validation, benchmarks, and schema validation
- **Cross-Vendor Examples**: 8 comprehensive pricing examples covering AWS, Azure, GCP, Kubernetes
- **Local Tool Management**: Automated buf CLI installation with version pinning
- **Comprehensive Documentation**: Complete testing guides and development workflows

### Testing Framework (`sdk/go/testing/`)
- **TestHarness**: In-memory gRPC testing with bufconn for reliable, network-free testing
- **MockPlugin**: Configurable mock implementation with error injection and custom behavior
- **Integration Tests**: Comprehensive test coverage for all 5 RPC methods
- **Performance Benchmarks**: Memory profiling and response time measurement
- **Conformance Testing**: Basic/Standard/Advanced certification levels for plugin validation
- **Complete Documentation**: Developer guide for plugin testing requirements

### CI/CD Pipeline (`.github/workflows/ci.yml`)
- **Automated SDK Generation**: buf CLI installation and protobuf code generation
- **Multi-Stage Validation**: Unit tests → Integration tests → Conformance tests → Benchmarks
- **Schema Validation**: JSON schema validation for all example specifications
- **Code Quality**: golangci-lint and buf linting with breaking change detection
- **Performance Tracking**: Benchmark result artifacts and coverage reporting
- **Supply Chain Security**: Pinned action versions and secure tool installation

### Cross-Vendor Examples (`examples/specs/`)
- **AWS Examples**: EC2 hourly billing, Lambda per-invocation, S3 tiered storage
- **Azure Examples**: SQL Database DTU billing, VM per-second pricing
- **GCP Examples**: Preemptible spot instances, Cloud Storage standard pricing
- **Kubernetes**: Namespace CPU billing via Kubecost integration
- **Complete Documentation**: Billing model explanations and usage patterns

### Local Tool Management
- **Automated buf Installation**: Version-pinned local installation in `bin/` directory
- **Dependency Isolation**: Avoid system-wide tool conflicts with local binaries
- **CI Integration**: Seamless tool management for both local and CI environments
- **Clean Targets**: Proper cleanup of generated files and local tools

### Documentation Improvements
- **Enhanced README**: Complete developer guide with quick start and examples
- **Updated CLAUDE.md**: Session learnings, best practices, and common issue solutions
- **Testing Documentation**: Comprehensive plugin development and testing workflows
- **Architecture Documentation**: Complete system overview and component descriptions

### Integration Testing
- ✅ All RPC methods tested with mock and real implementations
- ✅ Error injection and edge case handling verified
- ✅ Concurrent request handling validated
- ✅ Resource cleanup and connection management tested

### Performance Validation
- ✅ Benchmark tests for all operations with memory profiling
- ✅ Response time baselines established
- ✅ Concurrent load testing performed
- ✅ Memory leak detection completed

### Conformance Testing
- ✅ Basic conformance (core functionality) - 100% pass rate
- ✅ Standard conformance (production readiness) - 100% pass rate
- ✅ Advanced conformance (high performance) - 100% pass rate
- ✅ Cross-platform compatibility verified

### Schema Validation
- ✅ All 8 example specifications validate against JSON schema
- ✅ Cross-vendor billing model coverage verified
- ✅ Schema syntax validation passes
- ✅ Integration with Go SDK validation confirmed

### CI/CD Validation
- ✅ Complete pipeline execution verified
- ✅ Parallel job execution optimized
- ✅ Artifact collection and storage tested
- ✅ Breaking change detection functional

Fixes #5 #6 #7 #11

## Summary by CodeRabbit

- **New Features**
  - Go plugin testing framework (in-process harness, configurable mock plugin, multi-level conformance, and performance benchmarks); centralized pricing API for billing modes/providers; many PricingSpec example payloads for AWS, Azure, GCP, and Kubernetes.

- **Tests**
  - New integration, conformance, and benchmark suites covering RPCs, error cases, concurrency, performance, and data/validation scenarios.

- **Documentation**
  - Major docs overhaul: project README, developer quick start, testing framework guide, examples catalog, and CI guidance.

- **Chores**
  - New CI workflow, local tool management and Makefile targets, npm package metadata, linting configs, and updated gitignore.

### CODE REVIEW

**Comment by rshade** [comment]
@coderabbitai you need to ingest the .golangci-lint.yml before you make suggestions.

File: sdk/go/testing/harness.go [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+
+		metrics.TotalDuration += duration
+		if duration < metrics.MinDuration {
+			metrics.MinDuration = duration
+		}
+		if duration > metrics.MaxDuration {
+			metrics.MaxDuration = duration
+		}
+	}
+
```
[the remark this answers is not in the document]
Comment (rshade) [reply sdk/go/testing/harness.go:354]: This is correct in golang 1.24. Please update your codebase
